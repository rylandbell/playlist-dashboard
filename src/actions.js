export const getPlaylists = () => (
  {
    type: 'API',
    payload: {
      url: 'https://api.spotify.com/v1/me/playlists?limit=50',
      method: 'GET',
      success: 'GET_PLAYLISTS_SUCCESS',
      failure: 'GET_PLAYLISTS_FAILURE',
      pending: 'GET_PLAYLISTS_PENDING'
    }
  }
)

export const getTracks = (userId, playlistId) => (
  {
    type: 'API',
    payload: {
      url: `https://api.spotify.com/v1/users/${userId}/playlists/${playlistId}/tracks`,
      method: 'GET',
      success: 'GET_TRACKS_SUCCESS',
      failure: 'GET_TRACKS_FAILURE',
      pending: 'GET_TRACKS_PENDING'
    }
  }
)

export const getAudioFeatures = tracks => {
  const trackIds = tracks.map(track => track.track.id).join(',');
  return (
    {
      type: 'API',
      payload: {
        url: `https://api.spotify.com/v1/audio-features?ids=${trackIds}`,
        method: 'GET',
        success: 'GET_FEATURES_SUCCESS',
        failure: 'GET_FEATURES_FAILURE',
        pending: 'GET_FEATURES_PENDING'
      }
    }
  )
}

export const addPlaylistsData = (data) => (
  {
    type: 'ADD_PLAYLISTS_DATA',
    payload: data
  }
);

export const addTracksData = (data) => (
  {
    type: 'ADD_TRACKS_DATA',
    payload: data
  }
);

export const addFeaturesData = (data) => (
  {
    type: 'ADD_FEATURES_DATA',
    payload: data
  }
);

export const setAccessToken = data => (
  {
    type: 'SET_ACCESS_TOKEN',
    data: data
  }
);

export const selectPlaylist = (data, forceTabSwitch) => (
  {
    type: 'SELECT_PLAYLIST',
    forceTabSwitch: forceTabSwitch,
    data: data
  }
)

export const setActiveTab = data => (
  {
    type: 'SET_ACTIVE_TAB',
    data: data.substring(4)
  }
);

export const changeNameText = data => (
  {
    type: 'CHANGE_NAME_TEXT',
    data: data
  }
);

export const updateFilter = (filterIndex, inputValue) => (
  {
    type: 'UPDATE_FILTER',
    filterIndex: filterIndex,
    data: inputValue
  }
);

export const startDraggingFeatureSlider = filterIndex => (
  {
    type: 'START_DRAGGING_FEATURE_SLIDER',
    filterIndex: filterIndex
  }
);

export const stopDraggingFeatureSlider = () => (
  {
    type: 'STOP_DRAGGING_FEATURE_SLIDER'
  }
);

export const toggleChartedFeature = (filterIndex, newValue) => (
  {
    type: 'TOGGLE_CHARTED_FEATURE',
    filterIndex: filterIndex,
    newValue: newValue
  }
);

export const hoverOnTrack = track => (
  {
    type: 'HOVER_ON_TRACK',
    data: track
  }
);

export const clearHoveredTrack = () => (
  {
    type: 'CLEAR_HOVERED_TRACK',
  }
);

export const stopAnimatingChart = () => (
  {
    type: 'STOP_ANIMATING_CHART',
  }
);

export const badAuthToken = () => (
  {
    type: 'BAD_AUTH_TOKEN'
  }
);
