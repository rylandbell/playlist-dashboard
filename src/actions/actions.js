export const selectPlaylist = (data, forceTabSwitch) => ({
  type: "SELECT_PLAYLIST",
  forceTabSwitch: forceTabSwitch,
  data: data
});

export const setActiveTab = data => ({
  type: "SET_ACTIVE_TAB",
  data: data.substring(4)
});

export const changeNameText = data => ({
  type: "CHANGE_NAME_TEXT",
  data: data
});

export const sortByFeature = featureName => ({
  type: "SORT_BY_FEATURE",
  payload: {
    featureName: featureName
  }
});

export const updateFeatureFilter = (featureIndex, inputValue) => ({
  type: "UPDATE_FEATURE_FILTER",
  featureIndex: featureIndex,
  data: inputValue
});

export const startDraggingFeatureSlider = featureIndex => ({
  type: "START_DRAGGING_FEATURE_SLIDER",
  featureIndex: featureIndex
});

export const stopDraggingFeatureSlider = () => ({
  type: "STOP_DRAGGING_FEATURE_SLIDER"
});

export const toggleChartedFeature = (featureIndex, newValue) => ({
  type: "TOGGLE_CHARTED_FEATURE",
  featureIndex: featureIndex,
  newValue: newValue
});

export const hoverOnTrack = track => ({
  type: "HOVER_ON_TRACK",
  data: track
});

export const clearHoveredTrack = () => ({
  type: "CLEAR_HOVERED_TRACK"
});

export const stopAnimatingChart = () => ({
  type: "STOP_ANIMATING_CHART"
});
