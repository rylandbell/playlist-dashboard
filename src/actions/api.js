export const getPlaylists = () => ({
  type: 'API',
  payload: {
    url: 'https://api.spotify.com/v1/me/playlists?limit=50',
    method: 'GET',
    success: 'GET_PLAYLISTS_SUCCESS',
    failure: 'GET_PLAYLISTS_FAILURE',
    pending: 'GET_PLAYLISTS_PENDING'
  }
});

export const getTracks = (userId, playlistId) => ({
  type: 'API',
  payload: {
    url: `https://api.spotify.com/v1/users/${userId}/playlists/${playlistId}/tracks`,
    method: 'GET',
    success: 'GET_TRACKS_SUCCESS',
    failure: 'GET_TRACKS_FAILURE',
    pending: 'GET_TRACKS_PENDING'
  }
});

export const getAudioFeatures = tracks => {
  const trackIds = tracks.map(track => track.track.id).join(',');
  return {
    type: 'API',
    payload: {
      url: `https://api.spotify.com/v1/audio-features?ids=${trackIds}`,
      method: 'GET',
      success: 'GET_FEATURES_SUCCESS',
      failure: 'GET_FEATURES_FAILURE',
      pending: 'GET_FEATURES_PENDING'
    }
  };
};

export const createPlaylist = (userId, name) => ({
  type: 'API',
  payload: {
    url: `https://api.spotify.com/v1/users/${userId}/playlists`,
    method: 'POST',
    body: {
      name: name,
      public: false,
      collaborative: false
    },
    success: 'CREATE_PLAYLIST_SUCCESS',
    failure: 'CREATE_PLAYLIST_FAILURE',
    pending: 'CREATE_PLAYLIST_PENDING'
  }
});

export const addTracksToPlaylist = (userId, playlistId, tracksToSave) => {
  const trackURIArray = tracksToSave.map(track => track.uri);
  return {
    type: 'API',
    payload: {
      url: `https://api.spotify.com/v1/users/${userId}/playlists/${playlistId}/tracks`,
      method: 'POST',
      body: {
        uris: trackURIArray,
        position: 0
      },
      success: 'ADD_TRACKS_TO_PLAYLIST_SUCCESS',
      failure: 'ADD_TRACKS_TO_PLAYLIST_FAILURE',
      pending: 'ADD_TRACKS_TO_PLAYLIST_PENDING'
    }
  };
};
