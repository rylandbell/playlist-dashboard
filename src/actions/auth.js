export const authRequest = () => ({type: 'AUTH_REQUEST'});

export const setAccessToken = data => (
  {
    type: 'SET_ACCESS_TOKEN',
    data: data
  }
);