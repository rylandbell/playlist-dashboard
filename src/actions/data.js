export const addPlaylistsData = data => ({
  type: 'ADD_PLAYLISTS_DATA',
  payload: data
});

export const addTracksData = data => ({
  type: 'ADD_TRACKS_DATA',
  payload: data
});

export const addFeaturesData = data => ({
  type: 'ADD_FEATURES_DATA',
  payload: data
});
