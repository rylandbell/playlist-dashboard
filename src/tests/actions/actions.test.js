import * as actions from '../../actions/actions';

const testData = { data: 'test' };
const testFeatureIndex = 0;
const testFilterValue = [0, 1];

describe('General actions', () => {
  it('should create an action to select a playlist', () => {
    expect(actions.selectPlaylist(testData, true)).toMatchSnapshot();
  });

  it('should create an action to select a nav tab', () => {
    expect(actions.setActiveTab('testTabName')).toMatchSnapshot();
  });

  it('should create an action to change playlist name input text', () => {
    expect(actions.changeNameText('testInput')).toMatchSnapshot();
  });

  it("should create an action to update a feature filter's range value", () => {
    expect(
      actions.updateFeatureFilter(testFeatureIndex, testFilterValue)
    ).toMatchSnapshot();
  });

  it('should create an action to indicate that a feature slider is being dragged', () => {
    expect(
      actions.startDraggingFeatureSlider(testFeatureIndex)
    ).toMatchSnapshot();
  });

  it('should create an action to indicate that a feature slider has stopped being dragged', () => {
    expect(actions.stopDraggingFeatureSlider()).toMatchSnapshot();
  });

  it('should create an action to toggle whether a feature is drawn in chart', () => {
    expect(
      actions.toggleChartedFeature(testFeatureIndex, false)
    ).toMatchSnapshot();
  });

  it('should create an action to indicate that a track row is hovered', () => {
    expect(actions.hoverOnTrack(testData)).toMatchSnapshot();
  });

  it('should create an action to indicate that the cursor has left the tracks table', () => {
    expect(actions.clearHoveredTrack()).toMatchSnapshot();
  });

  it('should create an action to stop animating the chart', () => {
    expect(actions.stopAnimatingChart()).toMatchSnapshot();
  });
});
