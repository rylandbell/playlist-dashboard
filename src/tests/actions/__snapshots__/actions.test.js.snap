// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`General actions should create an action to change playlist name input text 1`] = `
Object {
  "data": "testInput",
  "type": "CHANGE_NAME_TEXT",
}
`;

exports[`General actions should create an action to indicate that a feature slider has stopped being dragged 1`] = `
Object {
  "type": "STOP_DRAGGING_FEATURE_SLIDER",
}
`;

exports[`General actions should create an action to indicate that a feature slider is being dragged 1`] = `
Object {
  "featureIndex": 0,
  "type": "START_DRAGGING_FEATURE_SLIDER",
}
`;

exports[`General actions should create an action to indicate that a track row is hovered 1`] = `
Object {
  "data": Object {
    "data": "test",
  },
  "type": "HOVER_ON_TRACK",
}
`;

exports[`General actions should create an action to indicate that the cursor has left the tracks table 1`] = `
Object {
  "type": "CLEAR_HOVERED_TRACK",
}
`;

exports[`General actions should create an action to select a nav tab 1`] = `
Object {
  "data": "TabName",
  "type": "SET_ACTIVE_TAB",
}
`;

exports[`General actions should create an action to select a playlist 1`] = `
Object {
  "data": Object {
    "data": "test",
  },
  "forceTabSwitch": true,
  "type": "SELECT_PLAYLIST",
}
`;

exports[`General actions should create an action to stop animating the chart 1`] = `
Object {
  "type": "STOP_ANIMATING_CHART",
}
`;

exports[`General actions should create an action to toggle whether a feature is drawn in chart 1`] = `
Object {
  "featureIndex": 0,
  "newValue": false,
  "type": "TOGGLE_CHARTED_FEATURE",
}
`;

exports[`General actions should create an action to update a feature filter's range value 1`] = `
Object {
  "data": Array [
    0,
    1,
  ],
  "featureIndex": 0,
  "type": "UPDATE_FEATURE_FILTER",
}
`;
